paper/src/main/java/vg/civcraft/mc/civmodcore/commands/CommandManager.java

```diff
@@ -31,25 +33,41 @@
  * Command registration class wrapper around {@link BukkitCommandManager}.
  */
 public class CommandManager extends BukkitCommandManager {
+	private final static List<String> allMaterials = Arrays.stream(Material.values()).map(Enum::name).toList();
+
+	private final static List<String> itemMaterials = Arrays.stream(Material.values()).filter(ItemUtils::isValidItemMaterial).map(Enum::name).toList();
+
+	// Track players to offer quick completion where necessary.
+	private final Set<String> autocompletePlayerNames = new ConcurrentSkipListSet<>();
 
 	private final CivLogger logger;
 
-    /**
+	/**
 	 * Creates a new command manager for Aikar based commands and tab completions.
 	 *
 	 * @param plugin The plugin to bind this manager to.
 	 */
-    public CommandManager(@Nonnull final Plugin plugin) {
-    	super(Objects.requireNonNull(plugin));
-    	this.logger = CivLogger.getLogger(plugin.getClass(), getClass());
-    }
+	public CommandManager(@Nonnull final Plugin plugin) {
+		super(Objects.requireNonNull(plugin));
+		this.logger = CivLogger.getLogger(plugin.getClass(), getClass());
+	}
 
 	/**
 	 * Will initialise the manager and register both commands and completions. You should only really use this if
 	 * you've used {@link CommandManager#reset()} or both {@link #unregisterCommands()} and
 	 * {@link #unregisterCompletions()}, otherwise there may be issues.
 	 */
-    public final void init() {
+	public final void init() {
+		// Prepare our list with player names on init.
+		Arrays.stream(Bukkit.getOfflinePlayers()).map(OfflinePlayer::getName).forEach(autocompletePlayerNames::add);
```

Comment:
Why is this in `init()` and not `registerCompletions()`?

Reply:
Not my code not my naming.

Reply:
Sure, but this *is* your PR which will likely be merged in lieu of Okx's #54 PR...

Reply:
I don't understand what the exact critizism is here, does the method have to be renamed or whats the issue?

Reply:
I've checked the original and both methods exist:
* https://github.com/CivMC/CivModCore/blob/00243ad09361361b46e808e3379eb739251a4365/paper/src/main/java/vg/civcraft/mc/civmodcore/commands/CommandManager.java#L52
* https://github.com/CivMC/CivModCore/blob/00243ad09361361b46e808e3379eb739251a4365/paper/src/main/java/vg/civcraft/mc/civmodcore/commands/CommandManager.java#L62

I've extended the init with a handler to update the existing player list, maybe I should've used a bit more comments and clarified the change here.

The idea is to load the offline players once and then keep them up to date with new login events, hence the async set. So updates and completions are cheap, but offline player completion behaviour is preserved.

If this is still an issue I'll ask you to be a bit more precise where exactly the issue lies in the method or the code.

Reply:
Comments added to clarify the intent here.